src/Microsoft.Health.Fhir.Core/Models/SearchParameterInfo.cs

```diff
@@ -56,11 +59,13 @@ public class SearchParameterInfo : IEquatable<SearchParameterInfo>
             Description = description;
         }
 
-        public SearchParameterInfo(string name)
+        public SearchParameterInfo(string name, string code)
         {
             EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
+            EnsureArg.IsNotNullOrWhiteSpace(code, nameof(code));
 
             Name = name;
+            Code = code;
         }
 
         public string Name { get; }
```

Comment:
>Name [](start = 22, length = 4)

Do we actually need it in this class?

Reply:
If we do, can I propose to add 
`public SearchParameterInfo(string nameAndCode): this(nameAndCode,nameAndCode){}`
constructor. 
code like this a 
`new SearchParameterInfo("a", "a")`  is not a prettiest one :D


---
In reply to: [568121485](https://github.com/microsoft/fhir-server/pull/1629#discussion_r568121485) [](ancestors = 568121485)

Reply:
Name is used a few times in messages to the end user where the more descriptive Name field may be more helpful than the Code.